### DISCUSSION TITLE
Can recoder pixiv sketch with login?

### DISCUSSION THREAD

**Discussion by a58958958** [lauwarm/docker-streamlink-recorder#27] [Q&A] [answered]
sorry my english  
 I tryed but didn't work 
maybe just i do it wrong?

**Comment by lauwarm** [comment, marked as answer]
Hey @a58958958 

I had a look at the Documentation of Streamlink and the Pixiv Plugin, which you can find here: https://streamlink.github.io/cli.html#cmdoption-pixiv-sessionid

It seems like a simple _username_ and _password_ login is not supported. Instead you need to use the **—pixiv-sessionid** and **—pixiv-devicetoken**.

To get those two you need to:
1. Go to pixiv and login with your account
2. Open a Livestream
3. Open Webdeveloper Tools
4. Find the Cookies

There you will find a **PHPSESSID** and **device_token.**

To run the Image with your specific pixiv-sessionid and pixiv-devicetoken you need to use the “-e streamOptions=“” option.

To give you an example:

```
docker run -v /path/to/vod/folder/:/home/download -e streamLink='' -e streamQuality='' -e streamName='' -e streamOptions='--pixiv-sessionid PHPSESSID --pixiv-devicetoken device_token' -e uid='' -e gid='' lauwarm/streamlink-recorder:latest
```

**Reply by a58958958** [reply]
it work! thank you verymuch !!
you are hero

**Reply by lauwarm** [reply]
Glad it works 👍
